scripts/src/javatests/org/oppia/android/scripts/maven/GenerateMavenDependenciesListTest.kt

```diff
@@ -0,0 +1,423 @@
+      )
+    }.build()
+    mavenDependencyList.writeTo(pbFile.outputStream())
+
+    val coordsList = listOf(DATA_BINDING_COORD, PROTO_LITE_COORD)
+    setupBazelEnvironment(coordsList)
+
+    GenerateMavenDependenciesListRunner(
+      mockLicenseFetcher,
+      CommandExecutorImpl(600_000L)
+    ).main(
+      arrayOf(
+        "${tempFolder.root}",
+        "scripts/assets/maven_install.json",
+        "${tempFolder.root}/scripts/assets/maven_dependencies.pb"
+      )
+    )
+    assertThat(outContent.toString()).contains(SCRIPT_PASSED_MESSAGE)
+  }
+
+  private fun setupBazelEnvironment(coordsList: List<String>) {
+    val mavenInstallJson = tempFolder.newFile("scripts/assets/maven_install.json")
+    writeMavenInstallJson(mavenInstallJson)
+    testBazelWorkspace.ensureWorkspaceIsConfiguredForRulesJvmExternal(coordsList)
+    val thirdPartyPrefixCoordList = coordsList.map { coordinate ->
+      "//third_party:${omitVersionAndReplaceColonsHyphensPeriods(coordinate)}"
+    }
+    createAndroidBinary(thirdPartyPrefixCoordList)
+    writeThirdPartyBuildFile(coordsList)
+  }
+
+  private fun writeThirdPartyBuildFile(exportsList: List<String>) {
+    val thirdPartyBuild = tempFolder.newFile("third_party/BUILD.bazel")
+    thirdPartyBuild.appendText(
+      """
+      load("@rules_jvm_external//:defs.bzl", "artifact")
+      """.trimIndent() + "\n"
+    )
+    for (export in exportsList) {
+      createAndroidLibrary(thirdPartyBuild, export)
+    }
+  }
+
+  private fun createAndroidLibrary(thirdPartyBuild: File, artifactName: String) {
+    thirdPartyBuild.appendText(
+      """
+      android_library(
+          name = "${omitVersionAndReplaceColonsHyphensPeriods(artifactName)}",
+          visibility = ["//visibility:public"],
+          exports = [artifact("$artifactName")],
+      )
+      """.trimIndent() + "\n"
+    )
+  }
+
+  private fun omitVersionAndReplaceColonsHyphensPeriods(artifactName: String): String {
+    val lastColonIndex = artifactName.lastIndexOf(':')
+    return artifactName.substring(0, lastColonIndex).replace('.', '_').replace(':', '_')
+  }
+
+  fun createAndroidBinary(
+    dependenciesList: List<String>
+  ) {
+    tempFolder.newFile("test_manifest.xml")
+    val build = tempFolder.newFile("BUILD.bazel")
+    build.appendText("depsList = [\n")
+    for (dep in dependenciesList) {
+      build.appendText("\"$dep\",")
+    }
+    build.appendText("]\n")
+    build.appendText(
+      """
+      android_binary(
+          name = "oppia",
+          manifest = "test_manifest.xml",
+          deps = depsList
+      )
+      """.trimIndent() + "\n"
+    )
+  }
+
+  private fun writeMavenInstallJson(file: File) {
+    file.writeText(
+      """
+      {
+        "dependency_tree": {
+          "dependencies": [
+            {
+              "coord": "androidx.databinding:databinding-adapters:3.4.2",
+              "url": "https://maven.google.com/androidx/databinding/databinding-adapters/3.4.2/databinding-adapters-3.4.2.aar"
```

Comment:
For these deps I suggest interpolating the constant variables defined above to keep the two places in sync.

Reply:
Done.